src/Sparrow.Server/Compression/Encoder3Gram.cs

```diff
@@ -428,94 +433,76 @@ private unsafe void BuildDictionary(in FastList<SymbolCode> symbolCodeList)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private unsafe int Lookup(ReadOnlySpan<byte> symbol, ReadOnlySpan<Interval3Gram> table, int numberOfEntries, out Code code)
+        private static unsafe int Lookup(ReadOnlySpan<byte> symbol, Interval3Gram* table, int numberOfEntries, out Code code)
         {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            static int CompareDictionaryEntry(ReadOnlySpan<byte> s1, ReadOnlySpan<byte> s2)
-            {
-                int length = s1.Length;
-                if (0 >= length)
-                    return s2[0] == 0 ? 0 : -1;
-                if (s1[0] != s2[0])
-                    return s1[0] < s2[0] ? -1 : 1;
-
-                if (1 >= length)
-                    return s2[1] == 0 ? 0 : -1;
-                if (s1[1] != s2[1])
-                    return s1[1] < s2[1] ? -1 : 1;
-
-                if (2 >= length)
-                    return s2[2] == 0 ? 0 : -1;
-                if (s1[2] != s2[2])
-                    return s1[2] < s2[2] ? -1 : 1;
-
-                return length > 3 ? 1 : 0;
-            }
+            // PERF: this is an optimized version of the CompareDictionaryEntry routine. Given that we
+            // can actually perform the operation in parallel. The usual case will be to call the parallel
+            // version instead of the serial, until we hit the end of the key. 
 
             int l = 0;
             int r = numberOfEntries;
 
+            uint symbolValue;
             if (symbol.Length >= 4)
             {
-                // PERF: this is an optimized version of the CompareDictionaryEntry routine. Given that we
-                // can actually perform the operation in parallel. The usual case will be to call the parallel
-                // version instead of the serial, until we hit the end of the key. 
-                uint symbolValue = BinaryPrimitives.ReverseEndianness(MemoryMarshal.Read<uint>(symbol)) >> 8;
-                while (r - l > 1)
-                {
-                    int m = (l + r) >> 1;
-
-                    uint codeValue = BinaryPrimitives.ReverseEndianness(table[m].BufferAndLength) >> 8;
-                    if (symbolValue < codeValue)
-                        r = m;
-                    else
-                        l = m;
-                }
+                symbolValue = BinaryPrimitives.ReverseEndianness(MemoryMarshal.Read<uint>(symbol)) >> 8;
             }
             else
             {
-                while (r - l > 1)
+                symbolValue = symbol.Length switch
```

Comment:
Please add a default path to eliminate warnings.

Reply:
Adding the default path requires throwing an exception; that cause the method not to be inlined and generate worse code. So it is not an option, best thing we can do is ignore the exception.